profileService/cmd/main.go

```diff
@@ -0,0 +1,16 @@
+package main
+
+import (
+	db2 "github.com/kas240sx89/samples/profileService/internal/db"
+	service2 "github.com/kas240sx89/samples/profileService/internal/service"
+)
+
+func main() {
+
+	database := db2.NewInMemoryDB()
+	var db service2.DB
+	db = &database
```

Comment:
These 3 lines can probably be condensed down to 1.

Reply:
this actually needs to be this way. this was the same issue we had before where we needed the interface to allow pointer receivers.